rcav2/models/errors.py

```diff
@@ -53,26 +57,39 @@ def read_target(target) -> str:
             return f"Unknown target: {target}"
 
 
-def read_error(anomaly) -> Error:
+def read_error(anomaly: dict, source: dict, logjuicer_url: str | None = None) -> Error:
+    """Creates an Error from an anomaly and sets the correct source URL."""
+    pos = anomaly["anomaly"]["pos"]
+    original_source_url = read_source(source)
+    final_url = original_source_url
+
+    if logjuicer_url and "TarFile" in source:
+        final_url = f"{logjuicer_url}#lr{pos}"
+
     return Error(
         before=anomaly["before"],
         line=anomaly["anomaly"]["line"],
-        pos=anomaly["anomaly"]["pos"],
+        pos=pos,
         after=anomaly["after"],
+        source=final_url,
     )
 
 
-def read_logfile(log_report) -> LogFile:
+def read_logfile(log_report: dict, logjuicer_url: str | None = None) -> LogFile:
+    source_json = log_report["source"]
     return LogFile(
-        source=read_source(log_report["source"]),
-        errors=list(map(read_error, log_report["anomalies"])),
+        source=read_source(source_json),
+        errors=[
+            read_error(anomaly, source_json, logjuicer_url)
+            for anomaly in log_report["anomalies"]
+        ],
     )
 
 
-def json_to_report(report) -> Report:
+def json_to_report(report: dict, logjuicer_url: str | None = None) -> Report:
     return Report(
         target=read_target(report["target"]),
-        logfiles=list(map(read_logfile, report["log_reports"])),
+        logfiles=[read_logfile(log, logjuicer_url) for log in report["log_reports"]],
```

Comment:
instead of resolving the full url here, we could add a `log_url` attribute to the report and get it from the json target value. That way we can resolve the URL in the react component.